ncm-ceph/src/main/perl/Ceph/compare.pm

```diff
@@ -268,39 +294,38 @@ sub compare_host {
         return 0; 
     } else {
         $self->compare_global($hostname, $quat_host->{config}, $ceph_host->{config}, $structures) or return 0;
-        if ($quat_host->{radosgw}) {
-            $self->compare_radosgw($hostname, $quat_host->{radosgw}->{config}, $ceph_host->{radosgw}->{config}, $structures);
-        } elsif ($ceph_host->{radosgw}) {
-            $self->info("radosgw config of $hostname not in quattor. Will get removed");
-        }
-        if ($quat_host->{mon} && $ceph_host->{mon}) {
-            $self->compare_mon($hostname, $quat_host->{mon}, $ceph_host->{mon}, $structures) or return 0;
-        } elsif ($quat_host->{mon}) {
-            $self->add_mon($hostname, $quat_host->{mon}, $structures) or return 0;
-        } elsif ($ceph_host->{mon}) {
-            $structures->{destroy}->{$hostname}->{mon} = $ceph_host->{mon};
-        }
-        if ($quat_host->{mds} && $ceph_host->{mds}) {
-            $self->compare_mds($hostname, $quat_host->{mds}, $ceph_host->{mds}, $structures) or return 0;
-        } elsif ($quat_host->{mds}) {
-            $self->add_mds($hostname, $quat_host->{mds}, $structures) or return 0;
-        } elsif ($ceph_host->{mds}) {
-            $structures->{destroy}->{$hostname}->{mds} = $ceph_host->{mds};
+        my @uniques = ('mon', 'mds');
+        foreach my $dtype (@uniques) {
+            if ($quat_host->{$dtype} && $ceph_host->{$dtype}) {
```

Comment:
maybe use 2 intermediate vars for `$quat_host->{$dtype}` `$ceph_host->{$dtype}` as they are used a lot